test/bin/manage_build_cache.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+#
+# This script should be run on the build host to manage cache of build artifacts
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck source=test/bin/common.sh
+source "${SCRIPTDIR}/common.sh"
+
+AWS_BUCKET_NAME="${AWS_BUCKET_NAME:-microshift-build-cache}"
+RELEASE_SUBDIR=
+BUILDID_SUBDIR=
+
+usage() {
+    cat - <<EOF
+${BASH_SOURCE[0]} (upload|download|exist|latest|keep) [options]
+
+Manage build cache at the '${AWS_BUCKET_NAME}' AWS S3 bucket. The script
+assumes that the bucket exists and it is configured for read-write access
+using 'aws s3 (ls|sync|rm)' operations.
+
+The cache directory structure is '${AWS_BUCKET_NAME}/<release>/<buildid>'.
+
+  -h        Show this help.
+
+  upload:   Upload build artifacts from the local disk to the specified
+            '${AWS_BUCKET_NAME}/<release>/<buildid>' AWS S3 bucket.
+
+  download: Download build artifacts from the specified
+            '${AWS_BUCKET_NAME}/<release>/<buildid>' AWS S3 bucket to
+            the local disk.
+
+  exist:    Exit with 0 status if the specified
+            '${AWS_BUCKET_NAME}/<release>/<buildid>' sub-directory exists
+            in a AWS S3 bucket, 1 otherwise.
+
+  latest:   Update the '${AWS_BUCKET_NAME}/<release>/latest' directory
+            in the AWS S3 bucket to a copy of the specified build ID.
+
+  keep:     Delete all data from the '${AWS_BUCKET_NAME}/<release>'
+            AWS S3 bucket, only keeping the '<buildid>' sub-directory.
+
+Options:
+
+  -r <release>  The release sub-directory in the cache.
+
+  -b <buildid>  The build ID sub-directory in the cache.
```

Comment:
What is `buildid`?

Reply:
I'll change this to `tag`. Is it better?

Reply:
Still, what is `tag`? to what it corresponds from local dir structure?